gguf-py/scripts/gguf-dump.py

```diff
@@ -101,25 +101,200 @@ def dump_metadata_json(reader: GGUFReader, args: argparse.Namespace) -> None:
+    markdown_content += '\n'
+
+    markdown_content += '| POS | TYPE       | Elements | Key                                    | Value                                                                          |\n'
+    markdown_content += '|-----|------------|----------|----------------------------------------|--------------------------------------------------------------------------------|\n'
+
+    for n, field in enumerate(reader.fields.values(), 1):
+        if not field.types:
+            pretty_type = 'N/A'
+        elif field.types[0] == GGUFValueType.ARRAY:
+            nest_count = len(field.types) - 1
+            pretty_type = '[' * nest_count + str(field.types[-1].name) + ']' * nest_count
+        else:
+            pretty_type = str(field.types[-1].name)
+
+        total_elements = len(field.data)
+        value = ""
+        if len(field.types) == 1:
+            curr_type = field.types[0]
+            if curr_type == GGUFValueType.STRING:
+                value = repr(str(bytes(field.parts[-1]), encoding='utf-8')[:60])
+            elif curr_type in reader.gguf_scalar_to_np:
+                value = str(field.parts[-1][0])
+        else:
+            if field.types[0] == GGUFValueType.ARRAY:
+                curr_type = field.types[1]
+                if curr_type == GGUFValueType.STRING:
+                    render_element = min(5, total_elements)
+                    for element_pos in range(render_element):
+                        value += repr(str(bytes(field.parts[-1 - element_pos]), encoding='utf-8')[:5]) + (", " if total_elements > 1 else "")
+                elif curr_type in reader.gguf_scalar_to_np:
+                    render_element = min(7, total_elements)
+                    for element_pos in range(render_element):
+                        value += str(field.parts[-1 - element_pos][0]) + (", " if total_elements > 1 else "")
+                value = f'[ {value}{" ..." if total_elements > 1 else ""} ]'
+        markdown_content += f'| {n:3} | {pretty_type:10} | {total_elements:8} | {field.name:38} | {value:<78} |\n'
+
+    markdown_content += "\n"
+
+    if not args.no_tensors:
+        # Group tensors by their prefix and maintain order
+        tensor_prefix_order = []
+        tensor_name_to_key = {}
+        tensor_groups = {}
+        total_elements = sum(tensor.n_elements for tensor in reader.tensors)
+
+        for key, tensor in enumerate(reader.tensors):
+            tensor_components = tensor.name.split('.')
+            tensor_prefix = tensor_components[0]
+
+            if tensor_prefix == 'blk':
+                tensor_prefix = f"{tensor_components[0]}.{tensor_components[1]}"
+
+            if tensor_prefix not in tensor_groups:
+                tensor_groups[tensor_prefix] = []
+                tensor_prefix_order.append(tensor_prefix)
+
+            tensor_name_to_key[tensor.name] = key
+            tensor_groups[tensor_prefix].append(tensor)
```

Comment:
I think top-level tensors which are not under `blk` should be grouped in the same group. (e.g. `output.weight`, `token_embd.weight`, `output_norm.weight`, etc.)

Reply:
Okay, decided to group it under a 'dummy' layer named 'base'.